### ISSUE TITLE
Weapon attachments no longer work

### ISSUE THREAD

**Issue by xLuxy** [altmp/altv-issues#1890]
### Description of the problem

When trying to add a weapon component to a weapon a player has equipped no attachments are actually applied.

### Reproduction steps

Run the following using QA-Tools / JS:
```js
player.giveWeapon('weapon_smg', 9999, true)
player.addWeaponComponent(player.currentWeapon, alt.hash('COMPONENT_AT_PI_SUPP'))

alt.log(player.hasWeaponComponent(player.currentWeapon, 'COMPONENT_AT_PI_SUPP')); // outputs: false
```

### Expected behaviour

Player should have suppressor attached to their SMG

### Additional context
`player.currentWeapon` has the correct value of the currently equipped weapon.

According to someone on [Discord](https://discord.com/channels/371265202378899476/371270983623114753/1127253705826250752) it was working 2 days ago.

### Operating system

Windows 11

### Version

dev/15.0-dev341

### Reproduction tested

- [X] I confirm that I have made sure that this issue is also present on the newest dev version

// Edit: The same was also tested in C#. Weapon component was still not added and HasWeaponComponent returned true while JS Module returned false

C# also returned false when an invalid weapon component was provided.

**Comment by xxshady** [comment]
By the way, reproduction is incorrect itself xd

`player.currentWeapon` will not return `weapon_smg` hash immediately, you can make sure of that with following code:
```js
// make sure no weapons are given before
player.removeAllWeapons(true)
await alt.Utils.wait(500)

player.giveWeapon('weapon_smg', 10, true)
// outputs: 2725352035 WEAPON_UNARMED
alt.log(player.currentWeapon, alt.getWeaponModelInfoByHash(player.currentWeapon).name)
```

Correct serverside reproduction code:
```js
const WEAPON = alt.hash('weapon_smg')
const COMPONENT = alt.hash('COMPONENT_AT_PI_SUPP')

// make sure no weapons are given before
player.removeAllWeapons(true)
await alt.Utils.wait(500)

player.giveWeapon(WEAPON, 10, true)
player.addWeaponComponent(WEAPON, COMPONENT)
 // outputs: true, but component is not set in game
alt.log(player.hasWeaponComponent(WEAPON, COMPONENT));
```

Confirmation that server-side code should work because native works:
```js
const WEAPON = alt.hash('weapon_smg')
const COMPONENT = alt.hash('COMPONENT_AT_PI_SUPP')

native.giveWeaponComponentToPed(alt.Player.local, WEAPON, COMPONENT)
```

![изображение](https://github.com/altmp/altv-issues/assets/54737754/d2118c56-899b-4582-80d3-6bcf574f29fd)

**Comment by xxshady** [comment]
Confirmed on 15.0-dev360

**Comment by xLuxy** [comment]
Issue has been fixed in 15.0-dev456
